trestlebot/tasks/sync_oscal_content_task.py

```diff
@@ -148,16 +179,63 @@ def write_ordered_data_to_yaml(
             yaml.indent(mapping=4, sequence=4, offset=4)
         yaml.dump(data, file_path)
 
-    def _update_selections_rules_in_memory(
-        self, rule_list: List[str], add_params: bool = True
+    def _update_control_file_change_in_memory(
+        self, cac_control: Dict[str, Any], oscal_control: ImplementedRequirement
+    ) -> None:
+        """
+        In memory update cac control file changes
+        """
+        rule_list = populate_if_dict_field_not_exist(cac_control, "rules", [])
+
+        removed_variable = []
+        cac_rule_list = []
+        for rule_index, rule in enumerate(rule_list):
+            if "=" in rule:
+                # variable
+                v_id, v_value = rule.split("=")
+                if v_id in self.parameter_diff_info.parameters_update:
+                    # update variable
+                    for v in self.parameter_diff_info.parameters_update[v_id]:
+                        rule_list[rule_index] = f"{v_id}={v}"
+                elif v_id in self.parameter_diff_info.parameters_remove:
+                    removed_variable.append(rule)
+            else:
+                # rule
+                cac_rule_list.append(rule)
+
+        # remove variables
+        for v in removed_variable:
+            rule_list.remove(v)
+
+        oscal_control_rules = [
+            prop.value for prop in oscal_control.props if prop.name == RULE_ID
+        ]
+        # remove rule
+        for rule in set(cac_rule_list).difference(set(oscal_control_rules)):
+            rule_list.remove(rule)
+
+        # add rule
+        for rule in set(oscal_control_rules).difference(set(cac_rule_list)):
+            if rule in self.all_rule_ids_from_cac:
+                rule_list.append(rule)
+            else:
+                logger.warning(
+                    f"rule {rule} not exists in cac content repo: {self.cac_content_root}"
+                )
+
+    def _update_profile_change_in_memory(
```

Comment:
There are similar code between `_update_profile_change_in_memory` and `_update_control_file_change_in_memory`. Have you considered to refactor in orders to use smaller functions for common code shared between other functions? For example, to parse variables.

Reply:
Already refactor this section. Make common code as function `_parse_single_variable`